cirq/work/work_pool_test.py

```diff
@@ -14,106 +14,117 @@
 
 import asyncio
 
+import pytest
+
 import cirq
 from cirq.work.work_pool import CompletionOrderedAsyncWorkPool
 
 
-def test_enqueue_then_dequeue():
+@pytest.mark.asyncio
+async def test_enqueue_then_dequeue():
     pool = CompletionOrderedAsyncWorkPool()
     work = asyncio.Future()
     pool.include_work(work)
     result = pool.__anext__()
-    cirq.testing.assert_asyncio_still_running(result)
+    assert await cirq.testing.asynchronous.asyncio_not_finishing(result)
```

Comment:
I was thinking `assert await foo` and `assert not await foo` are maybe ok, but I definitely think `assert await foo == bar` is  hard too parse. One possibility would be too absorb the assertion into the `cirq.testing.asynchronous` function, like in `numpy.testing.assert_allclose` and friends. Then you could just do
```python
await cirq.testing.asynchronous.assert_not_finishing(result)
```
You'd need a different function to assert that a future does finish, but maybe it's worth it. Might be good to get some other people's opinions about what would be least confusing here.

Reply:
Acknowledged. The function actually used to work that way. I changed it intentionally, to move the assert to the outside. IMO the only reason not to assert on the outside is for utility methods that give significantly better error messages on failure.